src/test/java/com/iexec/core/chain/BlockchainConnectionHealthIndicatorTests.java

```diff
@@ -0,0 +1,218 @@
+package com.iexec.core.chain;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.time.*;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class BlockchainConnectionHealthIndicatorTests {
+    private static final int POLLING_INTERVAL_IN_BLOCKS = 3;
+    private static final int OUT_OF_SERVICE_THRESHOLD = 4;
+    private static final Duration BLOCK_TIME = Duration.ofSeconds(5);
+    private static final Clock CLOCK = Clock.fixed(Instant.ofEpochSecond(1), ZoneId.systemDefault());
+
+    @Mock
+    private Web3jService web3jService;
+    @Mock
+    private ChainConfig chainConfig;
+    @Mock
+    private ScheduledExecutorService executor;
+
+    private BlockchainConnectionHealthIndicator blockchainConnectionHealthIndicator;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.openMocks(this);
+        when(chainConfig.getBlockTime()).thenReturn(BLOCK_TIME);
+
+        this.blockchainConnectionHealthIndicator = new BlockchainConnectionHealthIndicator(
+                web3jService,
+                chainConfig,
+                POLLING_INTERVAL_IN_BLOCKS,
+                OUT_OF_SERVICE_THRESHOLD,
+                executor,
+                CLOCK
+        );
+    }
+
+    // region scheduleMonitoring
+    @Test
+    void shouldScheduleMonitoring() {
+        blockchainConnectionHealthIndicator.scheduleMonitoring();
+
+        Mockito.verify(executor).scheduleAtFixedRate(
+                blockchainConnectionHealthIndicator.checkConnectionRunnable,
+                0L,
+                15L,
+                TimeUnit.SECONDS
+        );
+    }
+    // endregion
+
+    // region checkConnection
+
+    /**
+     * Returns a list of inputs, initial states and expected outputs for the `checkConnection` method
+     * in the following order:
+     * <ol>
+     *     <li>consecutiveFailures (initial state)</li>
+     *     <li>outOfService (initial state)</li>
+     *     <li>firstFailure (initial state)</li>
+     *     <li>latestBlockNumber (input)</li>
+     *     <li>consecutiveFailures (output)</li>
+     *     <li>outOfService (output)</li>
+     *     <li>firstFailure (output)</li>
+     * </ol>
```

Comment:
I think the javadoc best practice is more something like
```suggestion
     * <ol>
     * <li>consecutiveFailures (initial state)
     * <li>outOfService (initial state
     * <li>firstFailure (initial state)
     * <li>latestBlockNumber (input)
     * <li>consecutiveFailures (output)
     * <li>outOfService (output)
     * <li>firstFailure (output)
```

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-core/pull/601/commits/1b28e96e5859a2b84300736dde6fd7faaf416bef.

Reply:
Thanks :pray: